boa/network.py

```diff
@@ -164,6 +165,8 @@ def __init__(
 
         self._rpc: RPC = rpc
 
+        self.nickname = nickname or rpc.identifier
```

Comment:
`rpc.name` is more likely to strip sensitive info out of the url
```suggestion
        self.nickname = nickname or rpc.name
```